web-client/switch-api-colors.helpers.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  APIGatewayClient,
+  CreateBasePathMappingCommand,
+  DeleteBasePathMappingCommand,
+  GetRestApisCommand,
+} from '@aws-sdk/client-api-gateway';
+
+export const switchApiColors = async ({
+  deployingColor,
+  efcmsDomain,
+  environmentName,
+  publicApi,
+}: {
+  deployingColor: string;
+  efcmsDomain: string;
+  environmentName: string;
+  publicApi: boolean;
+}) => {
+  const regions = ['us-east-1', 'us-west-1'];
+  const apiGatewayRecordName = publicApi
+    ? `gateway_api_public_${environmentName}_${deployingColor}`
+    : `gateway_api_${environmentName}_${deployingColor}`;
+  const domainName = publicApi
+    ? `public-api.${efcmsDomain}`
+    : `api.${efcmsDomain}`;
+
+  for (const region of regions) {
+    const apiGateway = new APIGatewayClient({ region });
+    const getRestApisCommand = new GetRestApisCommand({ limit: 500 });
+    const { items } = await apiGateway.send(getRestApisCommand);
+
+    const apiGatewayRecord = items?.find(
+      record => record.name === apiGatewayRecordName,
+    );
+
+    if (apiGatewayRecord && 'id' in apiGatewayRecord && apiGatewayRecord.id) {
+      try {
+        const deleteBasePathMappingResult = await apiGateway.send(
+          new DeleteBasePathMappingCommand({
+            basePath: '(none)',
+            domainName,
+          }),
+        );
+        if (deleteBasePathMappingResult) {
+          const createBasePathMappingResult = await apiGateway.send(
+            new CreateBasePathMappingCommand({
+              domainName,
+              restApiId: apiGatewayRecord.id,
+              stage: environmentName,
+            }),
+          );
+          if (createBasePathMappingResult) {
+            console.log(createBasePathMappingResult);
+          }
+        }
+      } catch (err: any) {
```

Comment:
As a heads up this catch statement will log the error but swallow the error and have the script continue running. This means that the node script might error but not exit with a non zero status code. This is a bug we ran into in develop. This has been fixed in staging

Reply:
[Resolved](https://github.com/ustaxcourt/ef-cms/pull/4868/commits/8b3d51c353573dee8de35dd290ab7a06c1eaa2aa)